File: 10031-smallest-missing-integer-greater-than-sequential-prefix-sum/10031-smallest-missing-integer-greater-than-sequential-prefix-sum.java
class Solution {
    public int missingInteger(int[] nums) {
        int sum = nums[0];
        for(int i=1; i<nums.length; i++) {
            if(nums[i-1]+1 == nums[i]) {
                sum += nums[i];
            }
            else {
                break;
            }
        }
        
        HashSet<Integer> set = new HashSet<>();
        
        for(int i=0; i<nums.length; i++) {
            set.add(nums[i]);
        }
        
        int ans = sum;
        while(true) {
            if(!set.contains(sum)) {
                ans = sum;
                break;
            }
            else {
                sum += 1;
            }
        }
        
        return ans;
        
    }
}

File: 10031-smallest-missing-integer-greater-than-sequential-prefix-sum/10031-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        map<int,int>mp;
        int n=nums.size();
        for(int i=0;i<n;i++)
        {
            mp[nums[i]]++;
        }
        int sum=nums[0];
        for(int i=1;i<n;i++)
        {
            if(nums[i]==(nums[i-1]+1))
            {
                sum+=nums[i];
            }
            else break;
        }
        while(mp[sum]>0)
        {
            sum++;
        }
        return sum;
    }
};